azkaban-exec-server/src/test/java/azkaban/dag/DagTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.dag;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+import org.junit.Test;
+
+/**
+ * Tests the dag state ( including its nodes' states) transitions.
+ *
+ * Focuses on how the dag state changes in response to one external request.
+ */
+public class DagTest {
+
+  private final Dag testFlow = new Dag("fa", mock(DagProcessor.class));
+
+  @Test
+  public void dag_finish_with_only_disabled_nodes() {
```

Comment:
why is public test method name separated by _? should we follow a consistent standard and document in the contributor guide?

Reply:
Discussed offline, Cheng is fine with this style.